ext/stringio/stringio.c

```diff
@@ -1602,6 +1602,13 @@ strio_pread(int argc, VALUE *argv, VALUE self)
         rb_raise(rb_eArgError, "negative string size (or size too big): %" PRIsVALUE, rb_len);
     }
 
+    if (len == 0) {
+        if (NIL_P(rb_buf)) {
+            return rb_str_buf_new(0);
```

Comment:
Why do we need to use `rb_str_buf_new()` not `rb_str_new()` here?

Reply:
It simply seemed more natural to me, I'm happy to replace it by `rb_str_new("", 0)` if you prefer.

Reply:
OK. Then could you use `rb_str_new()` here because `IO#pread` uses `rb_str_new()` for this case?

Reply:
Of course. 

It's done.